### PR TITLE
Added pprof support to the Proxy Agent.

### PR DESCRIPTION
**Pull request by cheftako**
Added two enablement flags, both disabled by default.
1) --enable-profiling
2) --enable-contention-profiling

curl http://localhost:8094/debug/pprof/goroutine?debug=2  --output -

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -165,9 +176,13 @@ func (o *GrpcProxyAgentOptions) Validate() error {
 	if o.adminServerPort <= 0 {
 		return fmt.Errorf("admin server port %d must be greater than 0", o.adminServerPort)
 	}
+	if o.enableContentionProfiling && !o.enableProfiling {
+		return fmt.Errorf("if --enable-contention-profiling is set, --enable-profiling must also be set")
+	}
 	if o.syncInterval > o.syncIntervalCap {
 		return fmt.Errorf("sync interval %v must be less than sync interval cap %v", o.syncInterval, o.syncIntervalCap)
 	}
+
```
Comment (caesarxuchao) [inline cmd/agent/main.go:185]: nit: can you remove empty line?

File: pkg/util/handlers.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "net/http"
+
+// redirectTo redirects request to a certain destination.
```
Comment (caesarxuchao) [inline pkg/util/handlers.go:21]: nit: "RedirectTo redirects requests to a certain destination."

**Review by caesarxuchao — COMMENTED** [review]
Two nits. Otherwise lgtm.

File: pkg/util/handlers.go (hunk 2 above)
Comment (cheftako) [reply pkg/util/handlers.go:21]: Done

**Pull request merged**